parameter_delegate.cpp

```diff
@@ -161,38 +156,48 @@ namespace dealii
 
               return dirname_editor;
             }
-          else if (rx_integer.indexIn (pattern_description) != -1)		// if the tpye is "Integer"
+          else if (rx_integer.indexIn (pattern_description) != -1)		// if the type is "Integer"
             {
-              QSpinBox * spin_box = new QSpinBox(parent);			// choose a spin box
-
-              const int min_int_value = std::numeric_limits<int>::min();
-              const int max_int_value = std::numeric_limits<int>::max();
+              const QStringList default_pattern = pattern_description.split(" ").filter("...");
+              const QStringList default_values = default_pattern[0].split("...");
 
-              spin_box->setMaximum(max_int_value);				// set max and min from the limits.h class
-              spin_box->setMinimum(min_int_value);
-              spin_box->setSingleStep(int_steps);				// and every klick is a SingleStep
+              QLineEdit * line_edit = new QLineEdit(parent);
+              line_edit->setValidator(new QIntValidator(default_values[0].toInt(), default_values[1].toInt(), line_edit));
 
-              connect(spin_box, SIGNAL(editingFinished()),			// connect editors signal to the closer function
+              connect(line_edit, SIGNAL(editingFinished()),			// connect editors signal to the closer function
                       this, SLOT(commit_and_close_editor()));
 
-              return spin_box;
+              return line_edit;
             }
           else if (rx_double.indexIn (pattern_description) != -1)		// the same with "Double"
             {
-              QDoubleSpinBox * double_spin_box = new QDoubleSpinBox(parent);	// choose a spin box
-
-              const double min_double_value = -std::numeric_limits<double>::max();
-              const double max_double_value = std::numeric_limits<double>::max();
-
-              double_spin_box->setMaximum(max_double_value);		// set max and min from the limits.h class
-              double_spin_box->setMinimum(min_double_value);
-              double_spin_box->setDecimals(double_decimals);		// show "double_decimals" decimals
-              double_spin_box->setSingleStep(double_steps);		// and every klick is a SingleStep
-
-              connect(double_spin_box, SIGNAL(editingFinished()),		// connect editors signal to the closer function
+              const QStringList default_pattern = pattern_description.split(" ").filter("...");
+              QStringList default_values = default_pattern[0].split("...");
+
+              // Unfortunately conversion of MAX_DOUBLE to string and back fails
+              // sometimes, therefore use MAX_DOUBLE/2 to make sure we are below.
+              // In practice MAX_DOUBLE just means VERY large, it is normally not
+              // important how large.
+              const double max_double = std::numeric_limits<double>::max()/2;
+              const double min_double = std::numeric_limits<double>::min();
```

Comment:
Just noticed that MIN_DOUBLE is never used, `ParameterHandler` only uses MAX_DOUBLE and -MAX_DOUBLE. So no need for this.